### PR TITLE
Remove unnecessary kustomize in make helm

### LINKED ISSUE #1368: [Bug] make sync fails locally
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

**The bug:** When I ran make sync locally I got an error

```
test -s /vagrant/kuberay/ray-operator/bin/kustomize ||  GOBIN=/vagrant/kuberay/ray-operator/bin/kustomize/.. go install sigs.k8s.io/kustomize/kustomize/v3@v3.10.0
go: sigs.k8s.io/kustomize/kustomize/v3@v3.10.0 (in sigs.k8s.io/kustomize/kustomize/v3@v3.10.0):
        The go.mod file for the module providing named packages contains one or
        more exclude directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
Makefile:117: recipe for target 'kustomize' failed
make: *** [kustomize] Error 1

```
**Expected beahvior**
make sync should work fine and synchronize the required yamls.

**Useful Information**

Go version : go version go1.19.12 linux/amd64
Kustomize install version : v3@v3.10.0

My guess is that it is related to this issue: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-781521569

I could get around this issue by simply updating the kustomize version to 4.5.2 locally in the Makefile. Supporting comment for this: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-1054010950

I think the fix is just updating the version of Kustomize. However, I am not confident of what other things are dependent on this (Github CI/CD, other builds, etc). So, opening this issue for investigation. 

### Reproduction script

Navigate to `kuberay/ray-operator`

Run command : `make sync`

or 

`make helm`

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by shubhscoder**
## Why are these changes needed?

We first make kustomize as part of make helm (and hence make sync). But this is not required. This was realized while running make sync. This code change simply removes kustomize from make helm.

## Related issue number

Closes #1368 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by shubhscoder** [comment]
@kevin85421 This is the PR for 1368. I don't seem to have access to assignees section to add you as the reviewer. Thanks!

**Comment by kevin85421** [comment]
Could you remove kustomize from helm in the Makefile as I mentioned in https://github.com/ray-project/kuberay/issues/1368#issuecomment-1696777888? Additionally, would you test all the kustomize-related targets listed in the Makefile? In addition, are there any breaking changes between `3.10.0` and `5.1.1`?

**Comment by shubhscoder** [comment]
@kevin85421 Apologies for missing the helm part. I have updated the code changes.
Yes sure, I will test all the targets locally.
Also, let me investigate if there are any breaking changes from 3.10.0 to 5.1.1 

I will comment on this PR once I am done with the testing and the investigation. Thanks for the review.

File: ray-operator/Makefile [hunk 1]
```diff
@@ -114,7 +114,7 @@ controller-gen: ## Download controller-gen locally if necessary.
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
-	test -s $(KUSTOMIZE) ||  GOBIN=$(KUSTOMIZE)/.. go install sigs.k8s.io/kustomize/kustomize/v3@v3.10.0
+	test -s $(KUSTOMIZE) ||  GOBIN=$(KUSTOMIZE)/.. go install sigs.k8s.io/kustomize/kustomize/v5@v5.1.1
```
Comment (blublinsky) [inline ray-operator/Makefile:117]: Do you want to just use latest?

**Review by blublinsky — APPROVED** [review]
LGTM

File: ray-operator/Makefile (hunk 1 above)
Comment (shubhscoder) [reply ray-operator/Makefile:117]: @blublinsky v5.1.1 is the latest https://github.com/kubernetes-sigs/kustomize/releases?page=1

**Comment by shubhscoder** [comment]
@kevin85421 I tried the targets using Kustomize and they are building fine on my machine.
However, there are breaking changes that are introduced in v5@5.0.0 as seen [here](https://github.com/kubernetes-sigs/kustomize/releases?page=1). 
Therefore, I will try to investigate what of the breaking changes made are relevant to kuberay.

Also, the kustomize version being used in apiserver is 3.8.7, should we upgrade that as well?

**Comment by shubhscoder** [comment]
Closing this as the issue was fixed as part of https://github.com/ray-project/kuberay/commit/e7fbf7d73576d4f21671975709426645e5d38764

**Comment by shubhscoder** [comment]
Closing this as the issue was fixed as part of https://github.com/ray-project/kuberay/commit/e7fbf7d73576d4f21671975709426645e5d38764

**Comment by kevin85421** [comment]
> Also, the kustomize version being used in apiserver is 3.8.7, should we upgrade that as well?

^ cc @blublinsky

**Comment by shubhscoder** [comment]
> > Also, the kustomize version being used in apiserver is 3.8.7, should we upgrade that as well?
> 
> ^ cc @blublinsky

If needed, I can create a seperate PR for that. Since this one was open, I removed the unnecessary kustomize in make helm

**Comment by blublinsky** [comment]
Yes, I think we should upgrade kustomize throughout the project to the same version

**Review by kevin85421 — APPROVED** [review]

**Comment by shubhscoder** [comment]
> Yes, I think we should upgrade kustomize throughout the project to the same version


Thanks! I will create an issue and raise a CR to make kustomize version consistent across the project.

**Pull request merged**